### DISCUSSION TITLE
sim plot overlay

### DISCUSSION THREAD

**Discussion by zixingmun** [LAPKB/Pmetrics#217] [Q&A]
Hi everyone!

I would like to know if there is a way to overlay two sim plots on one graph that represent two different output equations `outeq`.

Here's my code for reference:

`sim1$plot(at = 1, outeq = 1, line = list(probs = c(0.25, 0.5, 0.75),` 
`                                         color = "black",`
`                                         dash = c("dashdot", "solid", "dashdot")),` 
`          legend = T, log = F, xlab = "Time, h", ylab = "Concentration, mg/L",` 
`          title = "Estimated Pharmacokinetic Profile", xlim = c(30, 36))`

`sim1$plot(at = 1, outeq = 2, line = list(probs = c(0.25, 0.5, 0.75),` 
`                                         color = "red",` 
`                                         dash = c("dashdot", "solid", "dashdot")),` 
`          legend = T, log = F, xlab = "Time, h", ylab = "Concentration, mg/L",` 
`          title = "Estimated Pharmacokinetic Profile", xlim = c(30, 36))`

Thanks in advance!

**Comment by mnneely** [comment]
Currently nothing simple, I’m afraid, but we could probably implement that in a future release. Currently,  Pmetrics allows differently formatted lines within the single outeq by specifying `color`, `dash`, or `width` as vectors. I don’t think we could allow that with multiple output equations plotted simultaneously, so if `outeq = c(1,2)` then the line formatting vectors would apply to each outeq as a whole. Something like this:

```
sim1$plot(outeq = c(1,2), line = list(probs = c(0.25, 0.5, 0.75),
                                         color = c(“black”,"red"),
                                         dash = c("dashdot", "solid")), #if only one value, would be recycled
          legend = T, log = F, xlab = "Time, h", ylab = "Concentration, mg/L",
          title = "Estimated Pharmacokinetic Profile", xlim = c(30, 36))
```

So all the quantile lines for outeq 1 would be black/dashdot, and all the lines for outeq 2 would be red/solid. You could not differentially format the median, for example, within each outeq. Would that be acceptable?

**Comment by zixingmun** [comment]
Hi Michael,

Thanks for your response.

I've attached my graphs for your reference with Plot 01 ![Plot 01](https://github.com/LAPKB/Pmetrics/assets/162079801/0a5b40c7-cdca-491e-82bb-e7c2cdc18155) being `outeq = 1` and Plot 02 ![Plot 02](https://github.com/LAPKB/Pmetrics/assets/162079801/9b6d64fe-147a-4624-82a0-8bc5f0a1fa7b) being `outeq = 2`.

I've tried the manual way of creating data frames for x and y values for both output equations using the 0.5 quantile and got the plot ![Plot 03](https://github.com/LAPKB/Pmetrics/assets/162079801/003c76e3-9bbe-47a6-84fe-737c95ee5831) I wanted.

When I tried to use the code you suggested, the curves ![Plot 04](https://github.com/LAPKB/Pmetrics/assets/162079801/a556d085-74ea-40af-8f3a-537203e6c738) ended up looking different, as in going up and down and I was wondering if you would know why that's the case? 

Also, if you don't mind me asking another question. I would like to plot the graph x-axis with time after dose instead of time. Here is my data template code for reference.

`dat <- PM_data$new()$`
`  addEvent(id = 1, time = 0, dose = 500, dur = 0, addl = 5, ii = 6)$`
`  addEvent(id = 1, time = 36, out = -1, outeq = 1)$`
`  addEvent(id = 1, time = 36, out = -1, outeq = 2, validate = TRUE)`

By doing `sim1$plot`, I find that I can only limit the x-axis from 30 to 36. Is there a way I can do time after dose so that the 30th hour can be equal to 0?

Thank you so much for your help. It's much appreciated.

**Reply by mnneely** [reply]
The code I suggested was for a future implementation. It would not work properly now indeed as it is combining the two outputs into one.


Plotting time after dose for simulations would be a little strange IMO as that is used typically to condense sparse sampling theoretically at different times but all at steady state, so that the concentrations can be superimposed within one dose interval. That's not the case with a simulation. So subtracting 30 from all the times would be a bit misleading I think. However, you can do it by overriding the tick labels:

```
sim1$plot(
line = list(probs = c(0.25, 0.5, 0.75),
   color = c(“black”,"red"),
   dash = c("dashdot", "solid")
), 
legend = T, log = F, xlab = "Time, h", ylab = "Concentration, mg/L",
title = "Estimated Pharmacokinetic Profile", 
xlim = c(30, 36)), 
xaxis = list(tickvals = 30:36, ticklabels = 0:6)
)
```
